### ISSUE TITLE
Astro compatibility pragma

### ISSUE THREAD

**Issue by Enteleform** [antfu/unplugin-icons#169]
# Issue

I've been using `unplugin-icons` & `unplugin-auto-import` with [Astro](https://github.com/withastro/astro), and the combination has worked very well when using a single renderer.

When enabling multiple renderers, Astro requires a pragma comment (*`/** @jsxImportSource react */`*) to process JSX/TSX files with the appropriate renderer. Since those comments don't exist in the virtual files generated by `unplugin-icons`, the dev server crashes immediately with a notification of missing renderer pragmas.

Is there a way to prepend these pragmas to the virtual file contents that are generated?

# Additional Info

I've tried injecting them during the `transform` hook, but it seems that Astro has already consumed the virtual files at this point and the error is still thrown (*even when using `{enforce:"pre"}`*). I had a similar issue with modifying `.astro` files during `transform`, and @jonathantneal helped me to resolve the issue by recommending that I perform the modifications during `load` instead.

However, that approach doesn't seem to work in this case, because I can't use `fs.readFileSync` as I did for the non-virtual `.astro` files. I searched a bit and it seems like this sort of thing might be handled in Rollup via `this.load(id)`, but it seems that Vite does not have this capability: https://github.com/vitejs/vite/issues/6810

**Comment by userquin** [comment]
@Enteleform you can add your plugin before the astro one, just go here and check the plugin `astro-fix-order`:
https://stackblitz.com/edit/github-d4rp6r-usdxm9?file=astro.config.mjs

You can also install the `vite-plugn-inspect` to check you're doing the correct changes (you have the dev dep configure on previous repro).

**Comment by Enteleform** [comment]
> @Enteleform you can add your plugin before the astro one, just go here and check the plugin `astro-fix-order`

Okay I'll try that out. Thanks for the tip!

**Comment by francoromanol** [comment]
Hey @Enteleform, did it work for you?

**Comment by Enteleform** [comment]
@francoromanol 

Yes, I got it to work. It's been a while so I don't remember exactly what it took to get it working, but I ended up not having to use the plugin reordering approach from `astro-fix-order`, as the `transform` approach ended up working at some point. Astro has had a ton of updates since I opened the issue, so it might have just been something that got changed on their side. Also, the Vite issue with `load` seems to  have been resolved as well.
